Sources/_Runtime/Utilities/RelativePointers.swift

```diff
@@ -0,0 +1,53 @@
+//===----------------------------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Argument Parser open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+protocol RelativePointer {
+  associatedtype Pointee
+
+  var offset: Int32 { get }
```

Comment:
Does this need to be part of the protocol interface?

Reply:
Huh, it seems I don't actually use `pointee(from:)` anywhere (which is the only default that needed offset information, unless we wanted a `isNull` or something). The more that I think about it, for ArgumentParser's use case we can get rid of this protocol entirely.